diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocMethods.kt

```diff
@@ -66,18 +76,40 @@ class KdocMethods : Rule("kdoc-methods") {
         isFixMode = autoCorrect
         emitWarn = emit
 
-        if (node.elementType == FUN && node.getFirstChildWithType(MODIFIER_LIST).isAccessibleOutside()) {
-            val config = KdocMethodsConfiguration(configRules.getRuleConfig(MISSING_KDOC_ON_FUNCTION)?.configuration
-                    ?: mapOf())
-            val isTestMethod = node.hasTestAnnotation() || node.isLocatedInTest(params.fileName!!.splitPathToDirs(), config.testAnchors)
-            if (!isTestMethod && !node.isStandardMethod() && !node.isSingleLineGetterOrSetter()) {
-                checkSignatureDescription(node)
+        if (node.elementType == FUN) {
+            if (node.hasChildOfType(KDOC)){
+                checkKDocParam(node)
+            }
+            if (node.getFirstChildWithType(MODIFIER_LIST).isAccessibleOutside()) {
+                val config = KdocMethodsConfiguration(configRules.getRuleConfig(MISSING_KDOC_ON_FUNCTION)?.configuration
+                        ?: mapOf())
+                val isTestMethod = node.hasTestAnnotation() || node.isLocatedInTest(params.fileName!!.splitPathToDirs(), config.testAnchors)
+                if (!isTestMethod && !node.isStandardMethod() && !node.isSingleLineGetterOrSetter()) {
+                    checkSignatureDescription(node)
+                }
             }
         } else if (node.elementType == KDOC_SECTION) {
             checkKdocBody(node)
         }
     }
 
+    private fun checkKDocParam(node: ASTNode) {
+        val functionParamList = node.findChildByType(VALUE_PARAMETER_LIST)!!
+                .getChildren(null)
+                .filter { it.elementType == VALUE_PARAMETER }
+                .mapNotNull { it.findChildByType(IDENTIFIER) }
+
+        val kDocParamList = node.findAllNodesWithSpecificType(KDOC_TAG_NAME)
+                .filter { it.text == "@${PARAM_KEYWORD}" }
```

Comment:
Take a look on how it's done in `getMissingParameters` method. Tags can be easily converted to psi objects to avoid comparison of strings.